src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -171,9 +169,27 @@ public void RemoveAt(int index)
 			if (child is Element element)
 				element.Parent = null;
 
-			InvalidateMeasure();
+			RemoveFromHandler(child);
+		}
+
+		void AddToHandler(IView view)
+		{
+			Handler?.Invoke(nameof(ILayoutHandler.Add), view);
+		}
+
+		void ClearHandler() 
+		{
+			Handler?.Invoke(nameof(ILayoutHandler.Clear));
+		}
 
-			LayoutHandler?.Remove(child);
+		void RemoveFromHandler(IView view)
+		{
+			Handler?.Invoke(nameof(ILayoutHandler.Remove), view);
+		}
+
+		void InsertIntoHandler(int index, IView view) 
+		{
+			Handler?.Invoke(nameof(ILayoutHandler.Insert), (index, view));
```

Comment:
With the very loose way tuples are, we probably should make some sort of type for this. Like the request for the scroll. This also is future proof where we can add new things.

And because we can have a strongly typed thing to cast to. This type can be used for all the commands:

```csharp
struct LayoutUpdatedArgs {
  IReadonlyList<IView> Added { get; set; }
  IReadonlyList<IView> Removed { get; set; }
  int InsertIndex { get; set; }
}
```

Reply:
Added a struct for passing the update information.